cartridges/int_algolia/cartridge/scripts/algolia/steps/algoliaProductDeltaIndex.js

```diff
@@ -359,25 +364,35 @@ exports.process = function(cpObj, parameters, stepExecution) {
                         let indexName = algoliaData.calculateIndexName('products', locale);
                         let records = recordsPerLocale[locale];
                         processedVariantsToSend = records.length;
+                        
                         records.forEach(function(record) {
-                            algoliaOperations.push(new jobHelper.AlgoliaOperation(baseIndexingOperation, record, indexName))
+                            if (INDEX_OUT_OF_STOCK || record.in_stock) {
+                                algoliaOperations.push(new jobHelper.AlgoliaOperation(baseIndexingOperation, record, indexName))
+                            } else {
+                                algoliaOperations.push(new jobHelper.AlgoliaOperation(deleteIndexingOperation, { objectID: record.objectID }, indexName));
+                            }
                         });
                     }
                 } else {
                     // Master-level indexing
-                    let baseModel = new AlgoliaLocalizedProduct({ product: product, locale: 'default', attributeList: nonLocalizedMasterAttributes });
-                    for (let l = 0; l < siteLocales.size(); ++l) {
-                        let locale = siteLocales[l];
-                        let indexName = algoliaData.calculateIndexName('products', locale);
-                        let localizedMaster = new AlgoliaLocalizedProduct({
-                            product: product,
-                            locale: locale,
-                            attributeList: masterAttributes,
-                            variantAttributes: variantAttributes,
-                            baseModel: baseModel,
-                        });
-                        processedVariantsToSend = localizedMaster.variants ? localizedMaster.variants.length : 0;
-                        algoliaOperations.push(new jobHelper.AlgoliaOperation(baseIndexingOperation, localizedMaster, indexName));
+                    let inStock = productFilter.isInStock(product, ALGOLIA_IN_STOCK_THRESHOLD);
```

Comment:
This can't work! As I explained in https://github.com/algolia/algoliasearch-sfcc-b2c/pull/218#discussion_r1923605183 your filtering function differs from the logic in `algoliaLocalizedProduct.js`
You had improved the logic yourself in https://github.com/algolia/algoliasearch-sfcc-b2c/pull/217 to take into account master products!

For master products, your `isInStock` function ALWAYS returns `false`
The consequence is that you ALWAYS end up in the `else` condition where `indexName` is not defined...

So here is what you would have seen if you had just run the delta job one time:

```
ERROR CustomJobThread|1976382993|AlgoliaProductDeltaIndex_v2|algoliaProductDeltaIndex
{"message":"No indexName found for this batch request near line:1 column:70","status":400}
```

Reply:
Really sorry for the back-and-forths and taking your time unnecessarily with these type errors. I will be more careful before requesting your review, and for each PR, I will create detailed test scenarios. Hopefully, it is working now. I also added my test scenarios; there is only two scenarios (similar) that we can think about, but I believe we can ignore them.

Please let me know what you think.